src/main/java/servlets/BillingConfig.java

```diff
@@ -15,47 +15,55 @@
 
 import data.JsonConverter;
 import data.Vendor;
-import data.Account;
+import util.FormIdNames;
+import static servlets.VendorServlet.updateSheets;
 
 import java.io.IOException;
 import java.security.GeneralSecurityException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import util.FormIdNames;
 
-import static servlets.VendorServlet.updateSheets;
 
 @WebServlet("/BillingConfig")
 public class BillingConfig extends HttpServlet {
   private static final String CONTENT_TYPE_APPLICATION_JSON = "application/json;";
-  private static final String REDIRECT_READFILE = "/index.html";
-
-  private Vendor vendor;
-  private Account account;
+  private static final String REDIRECT_INDEX = "/index.html";
 
   /**
-   * Read data endpoint.
+   * Read data endpoint for both the edit and read pages.
+   *
    * @param request form that contains valid vendor and account IDs
+   *                required fields: String vendorID & boolean entireConfig
    * @param response returns a configuration in JSON format
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String vendorID = request.getParameter(FormIdNames.VENDOR_ID);
     String accountID = request.getParameter(FormIdNames.ACCOUNT_ID);
+    // When getEntireConfig==true, return the entire configuration string.
+    boolean getEntireConfig = Boolean.parseBoolean(request.getParameter(
```

Comment:
I feel like this parseBoolean call should be within the try block below. I think this could throw an IOException if the caller did not put a boolean type in the ENTIRE_CONFIG param. Whether the error handing from the catch IOException block is sufficient for handling this error case, or there should be a new catch block to communicate this error state is up to you.

Reply:
It doesn't look like the parseBoolean method throws any error, but I went ahead and moved it down to the try block. The method just tests if the string == true && string != null, so I think everything else will set the boolean to false